hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -145,10 +145,14 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     OmBucketInfo omBucketInfo = null;
     if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
         .equals(BucketLayoutProto.LEGACY)) {
-      BucketLayout defaultType = BucketLayout.LEGACY;
-      if (StringUtils
-          .equalsIgnoreCase(OMConfigKeys.OZONE_OM_METADATA_LAYOUT_PREFIX,
-              omLayout)) {
+      // Bucket Layout argument was not passed during bucket creation.
```

Comment:
We can simply read the default value and assign to the bucket. Please remove the below if/else, which has some confusion.

```
// Bucket Layout argument was not passed during bucket creation.
      BucketLayout defaultType;
      String omDefaultBucketLayout = ozoneManager.getOMDefaultBucketLayout();
      if (StringUtils.equalsIgnoreCase(
          OMConfigKeys.OZONE_DEFAULT_BUCKET_LAYOUT_DEFAULT,
          omDefaultBucketLayout)) {
        defaultType = BucketLayout.OBJECT_STORE;
      } else {
        defaultType = BucketLayout.FILE_SYSTEM_OPTIMIZED;
``` 

Here the code can be as simple like,

```
    if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
        .equals(BucketLayoutProto.LEGACY)) {
                 // Bucket Layout argument is not passed during bucket creation, will assign OM default layout.
                 String omDefaultBucketLayout = ozoneManager.getOMDefaultBucketLayout();
                 BucketLayout defaultType = BucketLayout.fromString(String value);
                 omBucketInfo = OmBucketInfo.getFromProtobuf(bucketInfo, defaultType);
    }
```

Note: You have to implement `BucketLayout.fromString(String value);`, where you can do a string comparison and return matching enum type.


      if (StringUtils.equalsIgnoreCase(
          defaultBucketLayout, BucketLayout.OBJECT_STORE.name())) {
        defaultType = BucketLayout.OBJECT_STORE;
      } else if (StringUtils.equalsIgnoreCase(
          defaultBucketLayout, BucketLayout.FILE_SYSTEM_OPTIMIZED.name())) {
        defaultType = BucketLayout.FILE_SYSTEM_OPTIMIZED;
      } else {
          // Todo: need to discuss about throwing a validation error during OM startup or bucket creation time.
         defaultType = BucketLayout.LEGACY;
     }

Reply:
I've addressed the reviews in my latest patch, thanks!